packages/@sanity/form-builder/src/hooks/useDidUpdate.ts

```diff
@@ -0,0 +1,17 @@
+import {useEffect} from 'react'
+import {usePrevious} from './usePrevious'
+
+export function useDidUpdate<T>(value: T, didUpdate: (current: T, previous: T) => void): void
```

Comment:
This is a hard thing to name right. Maybe a more explicit naming could be `useOnChange`?

And the signature of the callback be similar to `shouldComponentUpdate`? e.g.
```
type OnChangeCallback<T> = (prev: T, next: T) => void
```

Reply:
I originally considered onChange but landed on useDidUpdate because onChange is usually about events coming from the UI whereas this one is "the value updated since previous render", so it more closely aligns with  componentDidUpdate semantics.

Agree on the order of callback params - your suggestion also aligns better with setState, reduce, etc. which expects the previous first.